src/core/parseHTML.js

```diff
@@ -15,7 +15,7 @@ define([
 // defaults to document
 // keepScripts (optional): If true, will include scripts passed in the html string
 jQuery.parseHTML = function( data, context, keepScripts ) {
-	if ( !data || typeof data !== "string" ) {
+	if ( typeof data !== "string" ) {
 		return null;
 	}
```

Comment:
> I don't think we should ever be returning null, despite the unit test.

This is the intent of my suggestion. Just dropping the `if` and letting values propagate through gives good (and mostly expected) results: [http://jsfiddle.net/evse952q/](http://jsfiddle.net/evse952q/), so we need not check for any special input or construct any special outptu (and it's worth noting that our internal uses always provide string input as documented).